src/DynamoPackages/PackageLoader.cs

```diff
@@ -382,6 +417,34 @@ public Package ScanPackageDirectory(string directory)
                     throw new LibraryLoadFailedException(directory, String.Format(Properties.Resources.NoHeaderPackage, headerPath));
                 }
 
+                // prevent loading unsigned packages if the certificates are required on package dlls
+                if (checkCertificates)
+                {
+                    foreach (var assemFile in (new System.IO.DirectoryInfo(discoveredPkg.BinaryDirectory)).EnumerateFiles("*.dll"))
+                    {
+                        try
+                        {
+                            var asm = Assembly.ReflectionOnlyLoadFrom(assemFile.FullName);
+                            var cert = asm.Modules.FirstOrDefault()?.GetSignerCertificate();
+                            if (cert != null)
+                            {
+                                var cert2 = new System.Security.Cryptography.X509Certificates.X509Certificate2(cert); 
+                                if(cert2.Verify())
+                                {
+                                    continue;
+                                }
+
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            //catch all exception types and pass to next throw
+                        }
+
+                        throw new LibraryLoadFailedException(directory, String.Format("A package called {0} found at {1} did not have signed dll files.  Ignoring it.", discoveredPkg.Name, discoveredPkg.RootDirectory));
```

Comment:
Shouldn't this be inside the above catch block?

Reply:
I can modify it to work that way.  I was trying to avoid two throws; one with the the try catch that covers any exception in the `Assembly.ReflectionOnlyLoadFrom` or cert creation and one I call directly because of the verify() result is `false`.  But it might be clearer code that way.